pubsub/gossipsub/gossipsub-v1.2.md

```diff
@@ -4,16 +4,103 @@
 |-----------------|---------------------------|--------|-----------------|
 | 1A              | Working Draft             | Active | r1, 2023-07-14  |
 
-Authors:
+Authors: [@Nashatyrev], [@Menduist]
 
-Interest Group: [@vyzo]
+Interest Group: [@vyzo], [@Nashatyrev], [@Menduist]
 
 [@vyzo]: https://github.com/vyzo
+[@Nashatyrev]: https://github.com/Nashatyrev
+[@Menduist]: https://github.com/Menduist
 
 See the [lifecycle document][lifecycle-spec] for context about maturity level and spec status.
 
 [lifecycle-spec]: https://github.com/libp2p/specs/blob/master/00-framework-01-spec-lifecycle.md
 
-## Overview
+# Overview
+
+This document aims to provide a minimal extension to the [gossipsub
+v1.1](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.1.md)
+protocol.
+
+The proposed extensions are backwards-compatible and aim to enhance the
+efficiency (minimize amplification/duplicates and decrease message latency) of
+the gossip mesh networks for larger messages. 
+
+In more specific terms, a new control message is introduced: `IDONTWANT`. It's primarily 
+intended to notify mesh peers that the node already received a message and there is no 
+need to send its duplicate.
+
+# Specification
+
+## Protocol Id
+
+Nodes that support this Gossipsub extension should additionally advertise the
+version number `1.2.0`. Gossipsub nodes can advertise their own protocol-id
+prefix, by default this is `meshsub` giving the default protocol id:
+- `/meshsub/1.2.0`
+
+## Parameters
+
+This section lists the configuration parameters that needs to agreed on across clients to avoid 
+ peer penalizations
+
+| Parameter                | Description                                                      | Reasonable Default |
+|--------------------------|------------------------------------------------------------------|--------------|
+| `max_idontwant_messages` | The maximum number of `IDONTWANT` messages per heartbeat per peer | ???  |
+
+
+## IDONTWANT Message
+
+### Basic scenario
+
+When the peer receives the first message instance it immediately broadcasts 
+(not queue for later piggybacking) `IDONTWANT` with the `messageId` to all its mesh peers. 
+This could be performed prior to the message validation to further increase the effectiveness of the approach.    
+
+On the other side a node maintains per-peer `dont_send_message_ids` set. Upon receiving `IDONTWANT` from 
+a peer the `messageId` is added to the `dont_send_message_ids` set. 
+When later relaying the `messageId` message to the mesh the peers found in `dont_send_message_ids` SHOULD be skipped. 
```

Comment:
Why SHOULD instead of MUST?

Reply:
In my mind since there is no way to enforce this behavior (after sending `IDONTWANT` a node may still receive the message from the correct peer) then there is basically no difference between `SHOULD` and `MUST`. 
I'm not sure about this though.

Reply:
yeah, if it cannot be enforced that it is a SHOULD.

Reply:
That sounds reasonable as well. 
I'm totally fine with either option

Reply:
@diegomrsantos your reasoning looks pretty fair. Thanks for comments 👍 
I've changed it to `MUST`.